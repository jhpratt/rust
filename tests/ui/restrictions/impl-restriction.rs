//@ compile-flags: --crate-type=lib

#![feature(impl_restriction)]

pub mod foo {
    pub mod bar {
        pub(crate) impl(super) trait Foo {}
    }

    impl bar::Foo for i8 {}
}

impl foo::bar::Foo for u8 {} //~ ERROR implementation of restricted trait
