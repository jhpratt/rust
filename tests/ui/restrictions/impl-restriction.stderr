error: implementation of restricted trait
  --> $DIR/impl-restriction.rs:13:1
   |
LL | impl foo::bar::Foo for u8 {}
   | ^^^^^^^^^^^^^^^^^^^^^^^^^
   |
note: trait restricted here
  --> $DIR/impl-restriction.rs:7:20
   |
LL |         pub(crate) impl(super) trait Foo {}
   |                    ^^^^^^^^^^^

error: aborting due to 1 previous error

